pdns/ws-auth.cc

```diff
@@ -445,21 +445,21 @@ static void fillZone(UeberBackend& backend, const DNSName& zonename, HttpRespons
     vector<DNSResourceRecord> records;
     vector<Comment> comments;
 
+    QType qType = QType::ANY;
+    DNSName qName;
+
     // load all records + sort
     {
       DNSResourceRecord resourceRecord;
       if (req->getvars.count("rrset_name") == 0) {
         domainInfo.backend->list(zonename, static_cast<int>(domainInfo.id), true); // incl. disabled
       }
       else {
-        QType qType;
-        if (req->getvars.count("rrset_type") == 0) {
-          qType = QType::ANY;
-        }
-        else {
+        qName = DNSName(req->getvars["rrset_name"]);
+        if (req->getvars.count("rrset_type") != 0) {
           qType = req->getvars["rrset_type"];
```

Comment:
Never a big fan of this implicit conversion, but ok.

Reply:
looking at this made me realise I'd prefer the throw-on-unknown-type variant (`DNSRecordContent::TypeToNumber`) here, but I don't want to backport that, so I'll clean up separately.

Reply:
the throwing version causes Internal Server Error, so that will require a bigger rework. Given that this is a search, I'll leave it at the current "nothing found" behaviour for now

Reply:
indeed, but from just seeing the assignment it's hard to figure out what happens for wrong types

Reply:
yes. might have been worth a comment.
not failing the request has a slight advantage for forward-compat, but 🤷🏼

Reply:
yeah, I also realised a comment would have been nice